src/java/spring/src/main/java/app/planentnine/springsessionless/adapter/web/UserController.java

```diff
@@ -0,0 +1,66 @@
+package app.planentnine.springsessionless.adapter.web;
+
+import app.planentnine.springsessionless.adapter.web.dto.RestUserDto;
+import app.planentnine.springsessionless.adapter.web.dto.mapper.RestUserDtoMapper;
+import app.planentnine.springsessionless.application.domain.User;
+import app.planentnine.springsessionless.application.port.incoming.CreateUserUseCase;
+import org.bouncycastle.jce.ECNamedCurveTable;
+import org.bouncycastle.jce.spec.ECNamedCurveParameterSpec;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import java.math.BigInteger;
+import java.util.HashMap;
+import java.util.Map;
+
+
+@Controller
+@RequestMapping("/api/user")
+public class UserController {
+    
+    private final CreateUserUseCase createUserUseCase;
+    private final RestUserDtoMapper userDtoMapper;
+    
+    @Autowired
+    public UserController(CreateUserUseCase createUserUseCase,
+                          RestUserDtoMapper restUserDtoMapper) {
+        this.createUserUseCase = createUserUseCase;
+        this.userDtoMapper = restUserDtoMapper;
+    }
+    
+    @PostMapping("/create")
+    public ResponseEntity<Object> createUser(@RequestBody RestUserDto restUserDto) {
+        User user = createUserUseCase.createUser(userDtoMapper.map(restUserDto));
+
+        if (isValidPublicKey(user.publicKey())) {
```

Comment:
So we should be verifying the signature here instead of validating the publicKey unless I'm reading this wrong. We do need to handle web too, but we can do that on a follow on, but if you do want to handle it you'll need to generate keys here and put the privateKey in a cookie. The JavaScript examples have that, but like I said we can do that in a follow up.

Reply:
Im not sure why we would attach a signature to the publicKey submission. Theres no extra data is there?
when you create a user youre just sending your public key to the database to be stored.
You could add extra content to user creation but for a minimilized implementation there isnt anything else to send

Reply:
You want to sign the timestamp to prevent replays to probe for uuids. It's an edge case, but it doesn't cost anything. I also think clients shouldn't have to worry about whether to sign or not, and just sign everything. 

Of course this _is_ an implementation so I guess it doesn't much matter since someone can just do this themselves 🤷‍♂️ . I think for consistency I'd say add it though.